src/config/backends/extra-uEnv.ts

```diff
@@ -0,0 +1,261 @@
+		return lines.reduce((options: ConfigOptions, line: string) => {
+			const optionValues = line.match(OPTION_REGEX);
+			if (optionValues == null) {
+				log.warn(`Could not read extra_uEnv entry: ${line}`);
+				return options;
+			}
+			// Merge new option with existing options
+			return {
+				...ExtraUEnvConfigBackend.parseOption(optionValues),
+				...options,
+			};
+		}, {});
+	}
+
+	private static parseOption(optionArray: string[]): ConfigOptions {
+		const [, KEY, VALUE] = optionArray;
+		// Check if this key's value is a collection
+		if (ExtraUEnvConfigBackend.entries[KEY as EntryKey]?.collection) {
+			// Return split collection of options
+			return ExtraUEnvConfigBackend.parseOptionCollection(VALUE);
+		}
+		// Find the option that belongs to this entry
+		const optionKey = _.findKey(
+			ExtraUEnvConfigBackend.supportedConfigs,
+			(_v, k) => ExtraUEnvConfigBackend.supportedConfigs[k].key === KEY,
+		);
+		// Check if we found a corresponding option for this entry
+		if (typeof optionKey !== 'string') {
+			log.warn(`Could not parse unsupported option: ${optionArray[0]}`);
+			return {};
+		}
+		return { [optionKey]: VALUE };
+	}
+
+	private static parseOptionCollection(
+		collectionString: string,
+	): ConfigOptions {
+		return (
+			collectionString
+				// Split collection into individual options
+				.split(' ')
+				// Reduce list of option strings into ConfigOptions object
+				.reduce((options: ConfigOptions, option: string) => {
+					// Match optionValues to key=value regex
+					const optionValues = option.match(OPTION_REGEX);
+					// Check if option is only a key
+					if (optionValues === null) {
+						if (option !== '') {
+							return { [option]: '', ...options };
+						} else {
+							log.warn(`Unable to set empty value option: ${option}`);
+							return options;
+						}
+					}
+					const [, KEY, VALUE] = optionValues;
+					// Merge new option with existing options
+					return { [KEY]: VALUE, ...options };
+				}, {})
+		);
+	}
+
+	private static async readBootConfigPath(): Promise<string> {
+		try {
+			return await fs.readFile(ExtraUEnvConfigBackend.bootConfigPath, 'utf-8');
+		} catch {
+			// In the rare case where the user might have deleted extra_uEnv conf file between linux boot and supervisor boot
+			// We do not have any backup to fallback too; warn the user of a possible brick
+			log.error(
+				`Unable to read extra_uEnv file at: ${ExtraUEnvConfigBackend.bootConfigPath}`,
+			);
+			throw new ExtraUEnvError(
+				'Could not find extra_uEnv file. Device is possibly bricked',
+			);
+		}
+	}
+
+	private static configToString(configs: ConfigOptions): string {
+		// Get Map of ConfigOptions object
+		const configMap = ExtraUEnvConfigBackend.configToMap(configs);
+		// Iterator over configMap and concat to configString
+		let configString = '';
+		for (const [key, value] of configMap) {
+			if (!value) {
+				log.warn(`Could not find value for ${key} in configMap.`);
```

Comment:
This shouldn't be possible, right?

Reply:
Very unlikely unless something had messed with the file. I was being extra verbose with this feature as I think more logs explaining why stuff doesn't happen is good.